packages/rpc-transport/src/json-rpc-transport/json-rpc-transport-types.ts

```diff
@@ -1,46 +1,74 @@
 /**
  * Public API.
  */
-export type Transport<TApi> = TransportMethods<TApi>;
-export type ArmedTransport<TApi, TResponse> = ArmedTransportMethods<TApi, TResponse> & {
-    send(): Promise<TResponse>;
+export type IRpcApi<TRpcMethods> = {
+    [MethodName in keyof TRpcMethods]: TRpcMethods[MethodName] extends Callable
+        ? (...rawParams: unknown[]) => TransportRequest<ReturnType<TRpcMethods[MethodName]>>
+        : never;
 };
-export type ArmedBatchTransport<TApi, TResponses extends unknown[]> = ArmedBatchTransportMethods<TApi, TResponses> & {
-    sendBatch(): Promise<TResponses>;
+export type Transport<TRpcMethods> = TransportMethods<TRpcMethods>;
+export type TransportConfig<TRpcMethods> = Readonly<{
+    api: IRpcApi<TRpcMethods>;
+    url: string;
+}>;
+export type TransportRequest<TResponse> = {
+    methodName: string;
+    params: unknown[];
+    responseProcessor?: (response: unknown) => TResponse;
```

Comment:
The whole universe of this PR starts from here. This is a brand new interface that you must comply with in order to supply RPC methods to a `@solana/rpc-transport`. Your concrete implementation of the RPC API is a bunch of methods that take in params, and return these things.